### ISSUE TITLE
Add binary use description to managers and lint

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#6438]
Managers that use third party binaries/images should have a section in their readme files that is then copied into the official docs.

**Comment by HonkingGoose** [comment]
Is there a list somewhere of those managers that rely on third party binaries/images?

Also what kind of message do we want to put in the section in their readme's?

I would take a stab at fixing this myself but I don't have the necessary information to start. 😄

**Comment by viceice** [comment]
You need to check all manager and if there is an `artifact.ts` then there will be a tool usage in it for most of them 🙃

**Comment by rarkins** [comment]
What I had in mind was to have a convention for describing artifacts generation, and that we add a custom lint check that looks whether a manager exports `updateArtifacts()` and if so then fails the test if the heading is not present in its README.

**Comment by HonkingGoose** [comment]
I like the idea of having a test to verify if we have the docs. 🚀 

So we will first need to decide on what the text in the heading should be. 😄

**Comment by rarkins** [comment]
Maybe something like:

```md
### Third party tool use

The `gomod` manager makes use of the `go` binary itself in order to run commands like `go get` and `go mod tidy`.

### Docker binary use

Renovate uses the [renovate/go](https://hub.docker.com/r/renovate/go) Docker Hub image. The images are tagged using the version of the `go` binary, e.g. `renovate/go:1.15.0`.
```

Although now I'm thinking most of the above could be expressed semantically instead of in MD :)

**Comment by rarkins** [comment]
I think this should work by checking which managers have `updateArtifacts` and then ensuring that they have the necessary documentation heading in their readme

**Comment by viceice** [comment]
No, terraform hat artifacts but we use a plain js implemention.

**Comment by rarkins** [comment]
Yes, but it can explain that in the same header. We just want to make sure that none *miss* having the docs.
